ui/src/app/components/help-sidebar-icons.tsx

```diff
@@ -0,0 +1,666 @@
+import { faDna } from '@fortawesome/free-solid-svg-icons/faDna';
+import { faLock } from '@fortawesome/free-solid-svg-icons/faLock';
+import { faSyncAlt } from '@fortawesome/free-solid-svg-icons/faSyncAlt';
+import { faTerminal } from '@fortawesome/free-solid-svg-icons/faTerminal';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import {
+  CdrVersionTiersResponse,
+  Criteria,
+  GenomicExtractionJob,
+  RuntimeStatus,
+  TerraJobStatus,
+} from 'generated/fetch';
+
+import colors, { colorWithWhiteness } from 'app/styles/colors';
+import { DEFAULT, reactStyles, switchCase } from 'app/utils';
+import { getCdrVersion } from 'app/utils/cdr-versions';
+import { ComputeSecuritySuspendedError } from 'app/utils/runtime-utils';
+import {
+  CompoundRuntimeOpStore,
+  RuntimeStore,
+  serverConfigStore,
+} from 'app/utils/stores';
+import { WorkspaceData } from 'app/utils/workspace-data';
+import { WorkspacePermissionsUtil } from 'app/utils/workspace-permissions';
+import { supportUrls } from 'app/utils/zendesk';
+import arrowLeft from 'assets/icons/arrow-left-regular.svg';
+import runtime from 'assets/icons/thunderstorm-solid.svg';
+import times from 'assets/icons/times-light.svg';
+import moment from 'moment/moment';
+
+import { RouteLink } from './app-router';
+import { FlexRow } from './flex';
+import { TooltipTrigger } from './popups';
+
+const styles = reactStyles({
+  asyncOperationStatusIcon: {
+    width: '.5rem',
+    height: '.5rem',
+    zIndex: 2,
+  },
+  runtimeStatusIconOutline: {
+    border: `1px solid ${colors.white}`,
+    borderRadius: '.25rem',
+  },
+  statusIconContainer: {
+    alignSelf: 'flex-end',
+    margin: '0 .1rem .1rem auto',
+  },
+  rotate: {
+    animation: 'rotation 2s infinite linear',
+  },
+  criteriaCount: {
+    position: 'absolute',
+    height: '0.8rem',
+    width: '0.8rem',
+    top: '1rem',
+    left: '0.55rem',
+    textAlign: 'center',
+    backgroundColor: colors.danger,
+    borderRadius: '50%',
+    display: 'inline-block',
+    fontSize: '0.4rem',
+  },
+  icon: {
+    background: colorWithWhiteness(colors.primary, 0.48),
+    color: colors.white,
+    display: 'table-cell',
+    height: '46px',
+    width: '45px',
+    borderBottom: `1px solid ${colorWithWhiteness(colors.primary, 0.4)}`,
+    cursor: 'pointer',
+    textAlign: 'center',
+    verticalAlign: 'middle',
+  },
+});
+
+const iconStyles = reactStyles({
+  active: {
+    ...styles.icon,
+    background: colorWithWhiteness(colors.primary, 0.55),
+  },
+  disabled: {
+    ...styles.icon,
+    cursor: 'not-allowed',
+  },
+});
+
+export interface IconConfig {
+  id: string;
+  disabled: boolean;
+  faIcon: IconDefinition;
+  label: string;
+  showIcon: () => boolean;
+  style: CSSProperties;
+  tooltip: string;
+  hasContent: boolean;
+}
+
+export const proIcons = {
+  arrowLeft: arrowLeft,
+  runtime: runtime,
+  times: times,
+};
+
+const displayRuntimeIcon = (
```

Comment:
this is the first of several **class methods** which were converted into **independent functions**